### PR TITLE
[CDF-22349] 🦮 Introduce Config

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cdf.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+featureflags = [
+    "import",
+]
+
+[modules]
+# This is the version of the modules. It should not be changed manually.
+# It will be updated by the 'cdf module upgrade' command.
+version = "0.2.18"
+
+[packages]
```
Comment (doctrino) [inline cdf.toml:16]: An alternative is to hardcode the packages into the CLI, however, I do not like that option.

File: cdf.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[cdf]
```
Comment (doctrino) [inline cdf.toml:1]:
```suggestion
[cdf]
project = "my_project"
```

File: cdf.toml (hunk 1 above)
Comment (doctrino) [inline cdf.toml:16]:
```suggestion
[modules.packages]
```

File: cdf.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+project_dir = "cognite_toolkit"
```
Comment (ronpal) [inline cdf.toml:2]: Should we consider another word than "project_dir"? I know we've used that until now, but project can be related to both a process and a CDF instance, none of which we are talking about here. How about work_dir or similar?

File: cdf.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "cdf_auth_readwrite_all",
+    "cdf_apm_base",
+    "cdf_infield_common",
+    "cdf_infield_location",
+]
+example_pump = [
+    "example_pump_asset_hierarchy",
+    "cdf_asset_source_model",
+    "example_pump_data_model",
+]
```
Comment (ronpal) [inline cdf.toml:32]: I think this is the way forward. Looking at the interactive init and the [manifest files](https://github.com/cognitedata/toolkit/blob/main/cognite_toolkit/_cdf_tk/prototypes/_packages/quickstart/manifest.yaml), the only thing missing (?) is a more explanatory text for each module to print in the interaction. But that can be solved in other ways. How about this? 
```suggestion
data_model_examples = {
    title: "Data model examples to learn from", 
    cdf_auth_readwrite_all: "Access groups that shouldn't be here",
    cdf_oid_example_data: "Actual sample data from the Valhall platform",
    cdf_apm_simple_data_model: "This is out of fashion anyway"
},
cdf_demo_infield = {
    title: "End-to-end setup of InField demo",
    cdf_auth_readwrite_all: "redundant groups",
    cdf_oid_example_data: "Actual sample data from the Valhall asset",
    cdf_apm_base: ""
    cdf_infield_common: "",
    cdf_infield_location: "",
}

...```

File: cdf.toml (hunk 3 above)
Comment (doctrino) [reply cdf.toml:2]: I agree, but I think we do not yet have a good alternative.

**Review by ronpal — APPROVED** [review]

**Pull request merged**